FileProviderExtensionUI/UploadProgressAlertController.swift

```diff
@@ -0,0 +1,94 @@
+//
+//  UploadProgressAlertController.swift
+//  FileProviderExtensionUI
+//
+//  Created by Philipp Schmid on 09.05.22.
+//  Copyright © 2022 Skymatic GmbH. All rights reserved.
+//
+
+import CryptomatorCommonCore
+import Promises
+import UIKit
+
+class UploadProgressAlertController: UIAlertController {
+	/// Promise fulfills as soon as a alert action gets triggered
+	let alertActionTriggered = Promise<Void>.pending()
+
+	var progress: Double? {
+		didSet {
+			DispatchQueue.main.async {
+				if let progress = self.progress {
+					self.setMessage(with: progress)
+				} else {
+					self.setMessageForMissingProgress()
+				}
+			}
+		}
+	}
+
+	private lazy var formatter: NumberFormatter = {
+		let formatter = NumberFormatter()
+		formatter.minimumFractionDigits = 0
+		formatter.maximumFractionDigits = 2
+		formatter.numberStyle = .decimal
+		return formatter
+	}()
+
+	func setMessage(with progress: Double) {
+		let formattedProgress = formatter.string(from: progress * 100.0 as NSNumber) ?? "n/a"
+		let text = "Current Progress: \(formattedProgress)%\n\nIf you're noticing that the upload progress is stuck, you can retry the upload."
```

Comment:
Requires localization.

Reply:
Oh, I actually forgot the entire localization :D